### ISSUE TITLE
Section 2.4.2 — Updated lambertian lacks new constructor

### ISSUE THREAD

**Issue by hollasch** [RayTracing/raytracing.github.io#1306]
### Discussed in https://github.com/RayTracing/raytracing.github.io/discussions/1301

<div type='discussions-op-text'>

<sup>Originally posted by **silvercorked** October 20, 2023</sup>
Book 2 chapter 4 section 2 concludes by passing a shared ptr of the newly created checker_texture class into to the make_shared<lambertian> function (Listing 25).
```
auto checker = make_shared<checker_texture>(0.32, color(.2, .3, .1), color(.9, .9, .9));
world.add(make_shared<sphere>(point3(0,-1000,0), 1000, make_shared<lambertian>(checker)));
// make_shared<lambertian>(checker) // lambertian has no constructor for shared_ptr<texture> by 4.2
```
An image is then shown, implying the code should be runnable. However, by this point, lambertian has no constructor that takes a shared_ptr<texture>. That constructor is introduced later in chapter 4, section 4 (Listing 30).
```
lambertian(shared_ptr<texture> a) : albedo(a) {}
```
It may be useful to introduce the constructor changes to the children of the material class before Listing 25, as walking through this awesome tutorial, I ended up having trouble here.</div>

**Comment by hollasch** [comment]
Looks like this is fixed in v4.0.0-alpha.1

**Comment by hollasch** [comment]
@bthtsang, please let me know if I'm mistaken.

**Comment by solimm4sks** [comment]
This is not fixed in v4.0.0-alpha1 from what I can see.
https://raytracing.github.io/books/RayTracingTheNextWeek.html#texturemapping/solidtextures:acheckertexture
The code is still in listing 30.
